src/service/inactive-administrative-cost-service.ts

```diff
@@ -0,0 +1,320 @@
+    await this.manager.save(InactiveAdministrativeCost, newInactiveAdministrativeCost);
+
+    const options = InactiveAdministrativeCostService.getOptions({ inactiveAdministrativeCostId: newInactiveAdministrativeCost.id });
+    return this.manager.findOne(InactiveAdministrativeCost, options);
+  }
+
+  /**
+   * Email all users with the given ids. These user will get notified that an administrative cost has been deducted from their account.
+   * @param users
+   */
+  public async handOutInactiveAdministrativeCost(users: HandoutInactiveAdministrativeCostsRequest)
+    : Promise<InactiveAdministrativeCost[]> {
+    return Promise.all(users.userIds.map(async (u) => {
+      const req: CreateInactiveAdministrativeCostRequest = { forId: u };
+
+      const inactiveAdministrativeCost = await this.createInactiveAdministrativeCost(req);
+
+      const user = await User.findOne({ where: { id: u } });
+
+      await Mailer.getInstance().send(user, new UserGotInactiveAdministrativeCost({
+        amount: inactiveAdministrativeCost.amount,
+      }));
+
+      return inactiveAdministrativeCost;
+    }));
+  }
+
+  /**
+   * Email all users with the given ids. These users will get notified that in a year time money will be deducted from their
+   * account as they have been inactive for three years.
+   * @param users
+   */
+  public async sendInactiveNotification(users: HandoutInactiveAdministrativeCostsRequest)
+    : Promise<void> {
+
+    await Promise.all(users.userIds.map(async (u) => {
+      const user = await User.findOne({ where: { id: u } });
+
+      user.inactiveNotificationSend = true;
+      await user.save();
+
+      return Mailer.getInstance().send(user, new InactiveAdministrativeCostNotification({}));
+    }),
+    );
+  }
+
+  /**
+   * Returns database entities based on the given filter params
+   * @param params
+   */
+  public async getInactiveAdministrativeCosts(params: InactiveAdministrativeCostFilterParameters = {})
+    : Promise<InactiveAdministrativeCost[]> {
+    const options = { ...InactiveAdministrativeCostService.getOptions(params) };
+    return this.manager.find(InactiveAdministrativeCost, { ...options });
+  }
+
+  /**
+   * Function that returns all inactive administrative cost entitites based on given params.
+   * @param params
+   * @param pagination - The pagination params to apply
+   */
+  public async getPaginatedInactiveAdministrativeCosts(params: InactiveAdministrativeCostFilterParameters = {},
+    pagination: PaginationParameters = {}) {
+    const { take, skip } = pagination;
+    const options = { ...InactiveAdministrativeCostService.getOptions(params), skip, take };
+
+    const inactiveAdministrativeCost = await this.manager.find(InactiveAdministrativeCost, { ...options, take });
+
+    const records = InactiveAdministrativeCostService.toArrayResponse(inactiveAdministrativeCost);
+
+    const count = await this.manager.count(InactiveAdministrativeCost, options);
```

Comment:
there is a find and count you can use

Reply:
This is the same way as in the other services which have been updated with extends WithManager